src/twenty-three-video/twenty-three-video.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { DisplayApiService } from "../display-api/display-api.service";
+import { HttpService } from "@nestjs/axios";
+import { Slide } from "../display-api/types";
+import { lastValueFrom } from "rxjs";
+import { SlideSlideInput } from "../display-api-client";
+
+interface TwentyThreeVideoListParams {
+  format: string;
+  raw: boolean;
+  tags?: string;
+  album_id?: string;
+}
+
+@Injectable()
+export class TwentyThreeVideoService {
+  constructor(
+    private readonly logger: Logger,
+    private readonly httpService: HttpService,
+    private readonly displayApi: DisplayApiService
+  ) {}
+
+  // TODO, move this out into configuration.
+  private readonly baseUrl = "https://video.kk.dk/api/photo/list";
+  private readonly slideType = "TwentyThreeVideo";
+
+  async syncAllSlides(): Promise<void> {
+    this.logger.debug("TwentyThreeVideoService::syncAllSlides");
+    const slides = await this.displayApi.fetchSlides(this.slideType);
+    for (const slide of slides) {
+      await this.syncSlide(slide);
+    }
+  }
+
+  async syncSlide(slide: Slide): Promise<void> {
+    const args: TwentyThreeVideoListParams = {
+      format: "json",
+      raw: true,
+    };
+    if (slide.content.tags) {
+      args.tags = slide.content.tags;
+    }
+    if (slide.content.album_id) {
+      args.album_id = slide.content.album_id;
+    }
+    const jsonData = await this.fetchTwentyThreeVideoList(args);
+    if (Array.isArray(jsonData)) {
+      slide.content.jsonData = JSON.stringify(jsonData);
```

Comment:
We should have some validation of this input. For a general approach I know @kasperg has his eyes on https://github.com/colinhacks/zod - we also have good experiences with https://github.com/pelotom/runtypes.

But, a simple "validateVideoList" function that throws or returns a SlideSlideInput will also do fine.

Reply:
This is really just checking if it's an empty value or not. I added a bit more data validation in `fetchTwentyThreeVideoList`